src/app/dashboard/user-administration/user-administration.component.ts

```diff
@@ -0,0 +1,140 @@
+import { Component, ViewChild, AfterViewInit, OnDestroy } from '@angular/core';
+import {
+  MatPaginator,
+  MatPaginatorModule,
+  PageEvent,
+} from '@angular/material/paginator';
+import { MatSort, MatSortModule } from '@angular/material/sort';
+import {
+  combineLatest,
+  debounceTime,
+  filter,
+  of as observableOf,
+  Subject,
+  takeUntil,
+} from 'rxjs';
+import { catchError, map, startWith, switchMap } from 'rxjs/operators';
+import { MatTableModule } from '@angular/material/table';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { DatePipe, TitleCasePipe } from '@angular/common';
+import { MatIcon } from '@angular/material/icon';
+import { MatButtonToggle } from '@angular/material/button-toggle';
+import { FormControl, FormsModule } from '@angular/forms';
+import {
+  MatFormField,
+  MatInput,
+  MatLabel,
+  MatPrefix,
+  MatSuffix,
+} from '@angular/material/input';
+import { MatButton, MatIconButton } from '@angular/material/button';
+import { RoundedInputComponent } from '@app/shared/rounded-input/rounded-input.component';
+import { UserAdministrationService } from '@app/dashboard/user-administration/user-administration.service';
+import { User } from '@app/auth/auth.type';
+import { MatChip } from '@angular/material/chips';
+import { MatMenu, MatMenuItem, MatMenuTrigger } from '@angular/material/menu';
+import { MatDialog } from '@angular/material/dialog';
+import { AddUserComponent } from '@app/dashboard/user-administration/add-user/add-user.component';
+
+@Component({
+  selector: 'app-user-administration',
+  styleUrl: 'user-administration.component.scss',
+  templateUrl: 'user-administration.component.html',
+  standalone: true,
+  imports: [
+    MatProgressSpinnerModule,
+    MatTableModule,
+    MatSortModule,
+    MatPaginatorModule,
+    DatePipe,
+    MatIcon,
+    MatButtonToggle,
+    FormsModule,
+    MatInput,
+    MatSuffix,
+    MatIconButton,
+    MatFormField,
+    MatLabel,
+    MatPrefix,
+    RoundedInputComponent,
+    MatButton,
+    MatChip,
+    TitleCasePipe,
+    MatMenuTrigger,
+    MatMenu,
+    MatMenuItem,
+  ],
+})
+export class UserAdministrationComponent implements AfterViewInit, OnDestroy {
+  displayedColumns: string[] = [
+    'created_at',
+    'name',
+    'email',
+    'role',
+    'status',
+    'more',
+  ];
```

Comment:
This could go inside the constants file

Reply:
It won't be used anywhere else, so not necessary